core/src/main/scala/akka/persistence/jdbc/journal/dao/DefaultJournalDao.scala

```diff
@@ -48,21 +52,34 @@ class DefaultJournalDao(
   val queries =
     new JournalQueries(profile, journalConfig.eventJournalTableConfiguration, journalConfig.eventTagTableConfiguration)
 
-  override def delete(persistenceId: String, maxSequenceNr: Long): Future[Unit] = {
-    val actions: DBIOAction[Unit, NoStream, Effect.Write with Effect.Read] = for {
-      _ <- queries.markJournalMessagesAsDeleted(persistenceId, maxSequenceNr)
-      highestMarkedSequenceNr <- highestMarkedSequenceNr(persistenceId)
-      _ <- queries.delete(persistenceId, highestMarkedSequenceNr.getOrElse(0L) - 1)
```

Comment:
This was indeed not efficient and there was also a little bug here. 

For a journal with events: 1, 2, 3, 4, 5

Deleting up to 3, would first:
logically delete: 1, 2, 3
collect the highest logically deleted nr, thus 3
delete everything until 3, so delete 1 and 2.

The journal remains with: 3, 4, 5 while 3 is invisible because logically deleted. 

There is no reason to keep 3, because it's not the highest anyway.